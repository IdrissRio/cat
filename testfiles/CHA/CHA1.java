// Example from http://web.cs.ucla.edu/~palsberg/tba/papers/dean-grove-chambers-ecoop95.pdf

class H extends F {}

class G extends F {}

class F extends C {
  @Override
  void p() {
  }
}

class E extends C {
  @Override
  void m() {
  }
}

class D extends B {
}

class C extends A {
  @Override
  void m() {
  }
}

class B extends A {
  @Override
  void m() {
  }
}

class A {
  void m() {
  }

  void p() {
  }
}