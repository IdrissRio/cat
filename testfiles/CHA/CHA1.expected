H :> []
H <: [A, C, F]
G :> []
G <: [A, C, F]
F :> [G, H]
F <: [A, C]
E :> []
E <: [A, C]
D :> []
D <: [A, B]
C :> [E, F, G, H]
C <: [A]
B :> [D]
B <: [A]
A :> [B, C, D, E, F, G, H]
A <: []

